src/authGuard.js

```diff
@@ -25,3 +32,46 @@ export const authGuard = (to, from, next) => {
         }
     });
 };
+
+/**
+ * Verifies the user is authenticated and has the provided permission. If the user is not authenticated, they will
+ * be redirected to the login page. If the user doesn't have the provided permission, the route change is not permitted.
+ * @param permission
+ * @return Exports function that returns a Vue router guard.
+ */
+export const permissionGuard = (permission) => {
+    return (to, from, next) => {
+        const authService = getInstance();
+
+        const fn = async () => {
```

Comment:
fn?

Reply:
https://auth0.com/docs/quickstart/spa/vuejs/01-login#secure-the-profile-page